connector-template/test.rs

```diff
@@ -22,70 +21,402 @@ impl utils::Connector for {{project-name | downcase | pascal_case}} {
+    assert_ne!(txn_id, None, "Empty connector transaction id");
+
+    // Capture
+    let capture_response = CONNECTOR
+        .capture_payment(txn_id.unwrap(), None, None)
+        .await
+        .unwrap();
+    assert_eq!(
+        capture_response.status,
+        enums::AttemptStatus::CaptureInitiated
+    );
 }
 
+// This test synchronizes a payment using the manual capture flow (Non 3DS).
 #[actix_web::test]
-async fn should_capture_already_authorized_payment() {
-    let connector = {{project-name | downcase | pascal_case}} {};
-    let authorize_response = connector.authorize_payment(None).await.unwrap();
+async fn should_synchronize_captured_payment() {
+    // Authorize
+    let authorize_response = CONNECTOR.authorize_payment(None, None).await.unwrap();
     assert_eq!(authorize_response.status, enums::AttemptStatus::Authorized);
     let txn_id = utils::get_connector_transaction_id(authorize_response);
-    let response: OptionFuture<_> = txn_id
-        .map(|transaction_id| async move {
-            connector.capture_payment(transaction_id, None).await.unwrap().status
-        })
-        .into();
-    assert_eq!(response.await.unwrap(), Some(enums::AttemptStatus::Charged));
+    assert_ne!(txn_id, None, "Empty connector transaction id");
+
+    // Sync
+    let sync_response = CONNECTOR
+        .sync_payment(
+            Some(types::PaymentsSyncData {
+                connector_transaction_id: router::types::ResponseId::ConnectorTransactionId(
+                    txn_id.unwrap(),
+                ),
+                encoded_data: None,
+                capture_method: None,
+            }),
+            None,
+        )
+        .await
+        .unwrap();
+    assert_eq!(sync_response.status, enums::AttemptStatus::Charged);
 }
 
+// This test void a payment using the manual capture flow (Non 3DS).
 #[actix_web::test]
-async fn should_fail_payment_for_incorrect_cvc() {
-    let response = {{project-name | downcase | pascal_case}} {}.make_payment(Some(types::PaymentsAuthorizeData {
-            payment_method_data: types::api::PaymentMethod::Card(api::CCard {
-                card_number: Secret::new("4024007134364842".to_string()),
-                ..utils::CCardType::default().0
+async fn should_void_manually_captured_payment() {
+    // Authorize
+    let authorize_response = CONNECTOR.authorize_payment(None, None).await.unwrap();
+    assert_eq!(authorize_response.status, enums::AttemptStatus::Authorized);
+    let txn_id = utils::get_connector_transaction_id(authorize_response);
+    assert_ne!(txn_id, None, "Empty connector transaction id");
+
+    // Void
+    let void_response = CONNECTOR
+        .void_payment(txn_id.unwrap(), None, None)
+        .await
+        .unwrap();
+    assert_eq!(void_response.status, enums::AttemptStatus::Voided);
+}
+
+// This test refund a payment using the manual capture flow (Non 3DS).
+#[actix_web::test]
+async fn should_refund_manually_captured_payment() {
```

Comment:
```suggestion
async fn should_refund_authorized_payment() {
```

Reply:
The name is to imply that this will test the refund scenario for manual capture flow. By naming "should_refund_authorized_payment" it would mean the test is for only authorised and not captured payment.